nikola/utils.py

```diff
@@ -1305,34 +1305,51 @@ class Commands(object):
     >>> commands.check(list=True)                # doctest: +SKIP
     """
 
-    def __init__(self, main):
+    def __init__(self, main, config, doitargs):
         """Takes a main instance, works as wrapper for commands."""
         self._cmdnames = []
-        for k, v in main.get_commands().items():
-            self._cmdnames.append(k)
+        self._main = main
+        self._config = config
+        self._doitargs = doitargs
+        for k, v in self._main.get_cmds().items():
```

Comment:
Note that `cmds` is a `PluginDict` [1], not a plain dict. You should get its items using `cmds.todict()`. Or if you need just one value, `cmds.get_plugin()`.

Plugins are lazy-loaded... for core commands plugins are not used so you get the command class without problems. But for plugins the dict value will only contain the entry point.


[1] https://github.com/pydoit/doit/blob/master/doit/plugin.py#L53